provisioning/transport/amqp/test/_amqp_test.js

```diff
@@ -446,5 +475,155 @@ describe('Amqp', function () {
     });
 
   });
+
+  describe('TPM', function() {
+    var fakeEndorsementKey = new Buffer('__FAKE_EK__');
+    var fakeStorageRootKey = new Buffer('__FAKE_SRK__');
+    var fakeAuthenticationKey = new Buffer('__FAKE_AUTH_KEY__');
+    var fakeSasToken = '__FAKE_SAS_TOKEN__';
+    var SaslTpm;
+
+    before(function() {
+      SaslTpm = sinon.spy(require('../lib/sasl_tpm'), 'SaslTpm');
+      sinon.stub(SaslTpm.prototype, 'getInitFrameContent');
+      sinon.stub(SaslTpm.prototype, 'getChallengeResponseContent');
+    });
+
+    after(function() {
+      SaslTpm.prototype.getInitFrameContent.restore();
```

Comment:
guessing we have to stub the prototype because we're instantiating the SaslTpm class within the Amqp transport and can't "inject" it?